src/foundry/foundry.js/collections/documentCollections/compendiumCollection.d.ts

```diff
@@ -0,0 +1,206 @@
+  class CompendiumCollection<T extends Metadata<any>> extends DocumentCollection<
+    DocumentClassForMetadata<T>,
+    'CompendiumCollection'
+  > {
+    constructor(metadata: T);
+
+    /** The compendium metadata which defines the compendium content and location */
+    metadata: T;
+
+    /**  A subsidiary collection which contains the more minimal index of the pack */
+    index: foundry.utils.Collection<DocumentInstanceForMetadata<T>>;
+
+    /**
+     * The amount of time that Document instances within this CompendiumCollection are held in memory.
+     * Accessing the contents of the Compendium pack extends the duration of this lifetime.
+     * @defaultValue `300`
+     */
+    static CACHE_LIFETIME_SECONDS: number;
+
+    /**
+     * The named game setting which contains Compendium configurations.
+     *  @defaultValue `compendiumConfiguration`
+     */
+    static CONFIG_SETTING: string;
+
+    /** The canonical Compendium name - comprised of the originating package and the pack name */
+    get collection(): string;
+
+    /** Access the compendium configuration data for this pack */
+    get config(): Record<string, CompendiumConfiguration>;
+
+    get documentName(): T['name'];
+
+    /** Track whether the Compendium Collection is locked for editing */
+    get locked(): boolean;
+
+    /** Track whether the Compendium Collection is private */
+    get private(): boolean;
+
+    /** A convenience reference to the label which should be used as the title for the Compendium pack. */
+    get title(): string;
+
+    get(key: string, { strict }: { strict: true }): DocumentInstanceForMetadata<T>;
+    get(key: string, { strict }?: { strict?: false }): DocumentInstanceForMetadata<T> | undefined;
+
+    set(id: string, document: DocumentInstanceForMetadata<T>): this;
+
+    delete(id: string): boolean;
+
+    /** Load the Compendium index and cache it as the keys and values of the Collection. */
+    getIndex(): Promise<DocumentInstanceForMetadata<T>['data']['_source']>;
+
+    /**
+     * Get a single Document from this Compendium by ID.
+     * The document may already be locally cached, otherwise it is retrieved from the server.
+     * @param id -  The requested Document id
+     * @returns The retrieved Document instance
+     */
+    getDocument(id: string): Promise<DocumentInstanceForMetadata<T> | undefined>;
+
+    /**
+     * Load multiple documents from the Compendium pack using a provided query object.
+     * @param query - A database query used to retrieve documents from the underlying database
+     * @returns The retrieved Document instances
+     */
+    getDocuments(query?: Query): Promise<DocumentInstanceForMetadata<T>[]>;
+
+    /**
+     * Import a Document into this Compendium Collection.
+     * @param document - The existing Document you wish to import
+     * @returns The imported Document instance
+     */
+    importDocument(document: DocumentInstanceForMetadata<T>): Promise<DocumentInstanceForMetadata<T>>;
```

Comment:
I guess this can also return `undefined`

Reply:
because of the error? I suppose so.

Reply:
Because of the changes that are made here to `Document.create`: https://github.com/League-of-Foundry-Developers/foundry-vtt-types/pull/880

Reply:
yup, @UFOMelkor is correct. The return types were changed because hooks may prevent document creation and in that case, the promise resolves to `undefined`.